kerl

```diff
@@ -606,62 +606,138 @@ assert_build_name_unused() {
             name=$(echo "$l" | cut -d, -f2)
             if [ "$name" = "$1" ]; then
                 error "there's already a build named '$1'."
-                exit 1
+                return 1
             fi
         done <"$KERL_BASE_DIR"/otp_builds
     fi
 }
 
+lock() {
+    # $1: action - open | close
+    # $2: lock type - build | install
+    # $3: folder to act on
+
+    if [ "$1" = "close" ]; then
+        if [ -f "$3/$2.lock" ]; then
+            error "trying to $2 in $3, but lock file ($3/$2.lock) exists!"
```

Comment:
i think if the lock file is "old" (like more than 14 days) we should remove it/ignore it

Reply:
Oh, yeah, I can implement that.

Reply:
2eef95c

Reply:
45e032c